apps/storefront/lib/auth/useSaleorAuthClient.ts

```diff
@@ -0,0 +1,42 @@
+import { SaleorAuthClient, SaleorAuthClientProps } from "./SaleorAuthClient";
+import { useEffect, useMemo, useState } from "react";
+
+export interface UseSaleorAuthClient {
+  saleorAuthClient: SaleorAuthClient;
+  isAuthenticating: boolean;
+}
+
+/**
+ * This hook should be used only once per application.
+ * @see Use {@link useSaleorAuthContext} for accessing the auth state.
+ */
+
+export const useSaleorAuthClient = ({
```

Comment:
Isn't this the same logic we've added for checkout?
Maybe it should be extracted into a separate package in monorepo?